naga/src/valid/mod.rs

```diff
@@ -565,6 +568,13 @@ impl Validator {
             if !decl_ty.equivalent(init_ty, gctx.types) {
                 return Err(OverrideError::InvalidType);
             }
+            self.validate_const_expression(init, gctx, mod_info, global_expr_kind)
+                .map_err(|err| OverrideError::ConstExpression {
+                    handle: init,
+                    source: err,
+                })?;
```

Comment:
Checking the initializer here is not needed since we already check all expressions of `global_expressions`.